app/[locale]/(interview)/interview/_components/Navigation.tsx

```diff
@@ -0,0 +1,112 @@
+import { useSearchParams } from 'next/navigation';
+import { usePathname, useRouter } from '~/lib/localisation/navigation';
+
+import { cn } from '~/lib/utils';
+import LanguageSwitcher from '~/app/_components/LanguageSwitcher';
+
+const NavigationButton = ({
+  onClick,
+  className,
+  ariaLabel,
+  children,
+}: {
+  onClick?: React.MouseEventHandler<HTMLButtonElement>;
+  className?: string;
+  ariaLabel?: string;
+  children: React.ReactNode;
+}) => {
+  return (
+    <button
+      className={cn(
+        `m-4 flex h-[4.8rem] w-[4.8rem] basis-[4.8rem] cursor-pointer items-center justify-center rounded-full transition-all`,
+        'hover:bg-primary',
+        className,
+      )}
+      tabIndex={0}
+      onClick={onClick}
+      aria-label={ariaLabel}
+    >
+      {children}
+    </button>
+  );
+};
+
+type NavigationProps = {
+  pulseNext: boolean;
+  progress: number;
+};
+
+const Navigation = ({ pulseNext, progress }: NavigationProps) => {
+  const router = useRouter();
+  const pathname = usePathname();
+  const searchParams = useSearchParams();
+  const stage = searchParams.get('stage');
+  const [expanded, setExpanded] = useState(false);
+
+  const moveBackward = () => {
+    if (!stage) return;
+    const newStage = parseInt(stage) - 1;
+    router.push(`${pathname}/?stage=${newStage}`);
+  };
+
+  const moveForward = () => {
+    if (!stage) return;
+    const newStage = parseInt(stage) + 1;
+    router.push(`${pathname}/?stage=${newStage}`);
+  };
+
+  const toggleMenu = () => {
+    setExpanded(!expanded);
+  };
+
+  const t = useTranslations('Navigation');
+
+  return (
+    <div
+      role="navigation"
+      className={cn(
+        'flex h-full flex-shrink-0 flex-grow-0 flex-col items-center bg-cyber-grape text-white',
+        expanded && 'w-full',
+      )}
+    >
+      <NavigationButton onClick={toggleMenu} ariaLabel={t('menu')}>
+        {expanded ? (
+          <X className="h-[2.4rem] w-[2.4rem]" strokeWidth="3px" />
+        ) : (
+          <Menu className="h-[2.4rem] w-[2.4rem]" strokeWidth="3px" />
+        )}
+      </NavigationButton>
+      {!expanded && (
+        <div className="flex h-full flex-col items-center justify-between">
+          <NavigationButton onClick={moveBackward} ariaLabel={t('back')}>
+            <ChevronUp className="h-[2.4rem] w-[2.4rem]" strokeWidth="3px" />
+          </NavigationButton>
+          {progress}
+          <NavigationButton
+            className={cn(
+              pulseNext && 'animate-pulse bg-success',
+              'hover:bg-success',
+            )}
+            onClick={moveForward}
+            ariaLabel={t('forward')}
+          >
+            <ChevronDown className="h-[2.4rem] w-[2.4rem]" strokeWidth="3px" />
+          </NavigationButton>
+        </div>
+      )}
+      {expanded && (
+        <div className="p-4">
+          <div>Select language </div>
+          <LanguageSwitcher />
+        </div>
+      )}
+    </div>
```

Comment:
Maybe it's just a personal preference but I'd use ternary here for rendering content based on `expanded`.
Ignore this if you think your approach is more readable

Reply:
resolved [dbbdc63](https://github.com/complexdatacollective/Studio/pull/30/commits/dbbdc637ab13683f4e7032dc41149434ecb54cbf)